src/simulation/simulation.jl

```diff
@@ -843,9 +843,9 @@ function execute!(sim::Simulation; kwargs...)
         error("Simulation status is invalid, you need to rebuild the simulation")
     end
     try
-        open_store(store_type, get_store_dir(sim), "w") do store
-            set_simulation_store!(sim, store)
-            Logging.with_logger(logger) do
+        Logging.with_logger(logger) do
```

Comment:
This was causing some log statements in the store to get dropped.

Reply:
ok, I am wondering why?

Reply:
The `with_logger` causes the simulation logger to be the global logger for its entire code block. The way it was left a gap: any log messages sent by the simulation store in its `close` execution were not covered in that code block, and so were left out of the simulation log file.